addons/web/static/tests/core/domain_selector_tests.js

```diff
@@ -1405,4 +1388,241 @@ QUnit.module("Components", (hooks) => {
+            string: "Definitions",
+            type: "properties_definition",
+        };
+        serverData.models.product.records[0].definitions = [
+            {
+                name: "selection_prop",
+                string: "Selection",
+                type: "selection",
+                selection: [
+                    ["abc", "ABC"],
+                    ["def", "DEF"],
+                ],
+            },
+        ];
+        const parent = await makeDomainSelector({
+            readonly: true,
+            domain: `[]`,
+        });
+        const toTest = [
+            {
+                domain: `[("properties.selection_prop", "=", False)]`,
+                text: `PropertiesSelection is not set `,
+            },
+            {
+                domain: `[("properties.selection_prop", "!=", False)]`,
+                text: `PropertiesSelection is set `,
+            },
+            {
+                domain: `[("properties.selection_prop", "=", "abc")]`,
+                text: `PropertiesSelection = ABC`,
+            },
+            {
+                domain: `[("properties.selection_prop", "=", expr)]`,
+                text: `PropertiesSelection = expr`,
+            },
+            {
+                domain: `[("properties.selection_prop", "!=", "abc")]`,
+                text: `PropertiesSelection != ABC`,
+            },
+        ];
+        for (const { domain, text } of toTest) {
+            await parent.set(domain);
+            assert.strictEqual(target.querySelector(".o_domain_leaf").textContent, text);
+        }
+    });
+
+    QUnit.test("many2one field (readonly)", async (assert) => {
+        const toTest = [
+            {
+                domain: `[("product_id", "=", 37)]`,
+                text: "Product = xphone",
+            },
+            {
+                domain: `[("product_id", "=", 2)]`,
+                text: "Product = Inaccessible/missing record ID: 2",
+            },
+            {
+                domain: `[("product_id", "!=", 37)]`,
+                text: "Product != xphone",
+            },
+            {
+                domain: `[("product_id", "=", False)]`,
+                text: "Product is not set ",
+            },
+            {
+                domain: `[("product_id", "!=", False)]`,
+                text: "Product is set ",
+            },
+            {
+                domain: `[("product_id", "in", [])]`,
+                text: "Product is in (  )",
+            },
+            {
+                domain: `[("product_id", "in", [41, 37])]`,
+                text: "Product is in ( xpad , xphone )",
+            },
+            {
+                domain: `[("product_id", "in", [1, 37])]`,
+                text: "Product is in ( Inaccessible/missing record ID: 1 , xphone )",
+            },
+            {
+                domain: `[("product_id", "in", [1, uid, 37])]`,
+                text: 'Product is in ( "Inaccessible/missing record ID: 1" , uid , "xphone" )',
```

Comment:
Don't we want it to be `Product is in ( Inaccessible/missing record ID: 1 , uid , "xphone" )`

Reply:
I don't think so, the value "Inaccessible/missing record ID: 1" is a string. The quotes are there to distinguish
false and "false"
1 and "1"
"" and (nothing) 
...